app/beacon-omop-worker/beacon_omop_worker/query_solvers.py

```diff
@@ -307,3 +309,203 @@ def solve_individuals(db_manager: SyncDBManager, query_terms: str):
         return response_summary
     except Exception as e:
         logger.error(str(e))
+
+class KaplanMeierQuerySolver:
+
+    def __init__(self, db_manager) -> None:
+        self.db_manager = db_manager
+
+    def _get_concept_id(self, vocab: str, concept_code: str) -> str:
+        """
+        Get concept_id from vocabulary and concept code.
+        
+        Args:
+            vocab (str): The vocabulary name (e.g., 'SNOMED').
+            concept_code (str): The concept code (e.g., '386661006').
+        
+        Returns:
+            str: The concept_id corresponding to the vocabulary and concept code.
+        """
+        logger = logging.getLogger(config.LOGGER_NAME)
+        try:
+            query = select(Concept.concept_id).where(
+                and_(
+                    Concept.vocabulary_id == vocab,
+                    Concept.concept_code == concept_code
+                )
+            )
+            result = pd.read_sql(query, con=self.db_manager.engine.connect())
+            concept_id = str(result["concept_id"].values[0])
+            logger.info(f"Found concept_id {concept_id} for {vocab}:{concept_code}.")
+            return concept_id
+        except Exception as e:
+            logger.error(f"Error fetching concept_id for {vocab}:{concept_code}: {e}")
+            raise
+
+    def fetch_condition_data(self, concept_id: str) -> pd.DataFrame:
+        """
+        Fetch data from the OMOP database based on concept_id.
+        
+        Args:
+            concept_id (str): The concept_id for the condition.
+        
+        Returns:
+            pd.DataFrame: DataFrame containing the condition data.
+        """
+        logger = logging.getLogger(config.LOGGER_NAME)
+        
+        try:
+            # Load individual tables into pandas DataFrames with filtering
+            condition_occurrence_df = pd.read_sql(
```

Comment:
We have a table defined in `entities.py` for condition occurrences. You can actually create this query programmatically, similar to how it's done in `_get_concept_id`. Also you can check out the docs for sqlalchemy2 for extra help.